### PR TITLE
Bump docker dependency

### PR DESCRIPTION
**Pull request by xpivarc**
Fix a "invalid Host Header" error with
few Go versions

### CODE REVIEW

**Review by brianmcarey — COMMENTED** [review]
/lgtm

@xpivarc could you sign your commit?

**Comment by xpivarc** [comment]
@brianmcarey Done, thanks!

File: cluster-provision/gocli/go.mod [hunk 1]
```diff
@@ -19,28 +19,29 @@ require (
 	k8s.io/apimachinery v0.20.6
 )
 
+require (
```
Comment (brianmcarey) [inline cluster-provision/gocli/go.mod:22]: @xpivarc why are these not included with the rest of the indrect deps below?

Comment (xpivarc) [reply cluster-provision/gocli/go.mod:22]: Good question. I used standard tooling with no manual changes, therefore I am not sure.

Comment (brianmcarey) [reply cluster-provision/gocli/go.mod:22]: Seems to be a common issue - https://github.com/golang/go/issues/56471

Not sure if it is worth manually tidying this?

**Review by brianmcarey — APPROVED** [review]
/approve

**Comment by brianmcarey** [comment]
@xpivarc - it looks like this change broke the publish job - https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/logs/publish-kubevirtci/1697339447776907264

Not sure how it wasn't caught by the check-provisions?

**Comment by xpivarc** [comment]
> @xpivarc - it looks like this change broke the publish job - https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/logs/publish-kubevirtci/1697339447776907264
> 
> Not sure how it wasn't caught by the check-provisions?

I am not quite sure as well. Let's revert or fix.

**Comment by dhiller** [comment]
> > @xpivarc - it looks like this change broke the publish job - https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/logs/publish-kubevirtci/1697339447776907264
> > Not sure how it wasn't caught by the check-provisions?
> 
> I am not quite sure as well. Let's revert or fix.

I think that this looks like a [quite old go version](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/logs/publish-kubevirtci/1697339447776907264#1:build-log.txt%3A8), no? Maybe rather update that one?

**Comment by oshoval** [comment]
> Not sure how it wasn't caught by the check-provisions?

publish is only postsubmit, check-provision is only presubmit

**Comment by oshoval** [comment]
Please note that this PR cause problems when i try to spin cluster locally,
building without it locally fixed the problem

more info 
https://github.com/kubevirt/kubevirt/pull/10375#issuecomment-1712803145

**Comment by brianmcarey** [comment]
> Please note that this PR cause problems when i try to spin cluster locally, building without it locally fixed the problem
> 
> more info [kubevirt/kubevirt#10375 (comment)](https://github.com/kubevirt/kubevirt/pull/10375#issuecomment-1712803145)

@oshoval are you trying this in podman or docker?

**Comment by oshoval** [comment]
> > Please note that this PR cause problems when i try to spin cluster locally, building without it locally fixed the problem
> > more info [kubevirt/kubevirt#10375 (comment)](https://github.com/kubevirt/kubevirt/pull/10375#issuecomment-1712803145)
> 
> @oshoval are you trying this in podman or docker?

pure docker, also the rebuilt with docker

**Pull request merged**